include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -74,6 +77,81 @@
 DEAL_II_NAMESPACE_OPEN
 
 
+namespace
+{
+  template <int dim, int fe_degree, int components, typename number>
+  class MassOperator : public Subscriptor
```

Comment:
i would move this to `matrix_free/operators.h` and add a unit test to compare with, say, Trilinos.